pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4149,7 +4157,7 @@ def _get_offset(name: str) -> BaseOffset:
     return _offset_map[name]
 
 
-cpdef to_offset(freq):
+cpdef to_offset(freq, is_period=None):
```

Comment:
Thanks, I made the changes that you suggested. Then in order to fix “errors compiling Cython files” I made the next modifications:

1. added in `timestamps.pyx` the parameter `is_period` into `to_offset(freq, is_period)` and `_round()`
2. in the file `offsets.pyx` I had an error `“Call with a wrong number of arguments (expected 2, got 1)”` in the method `__eq__`.  I defined a new method `to_offset_eq(...)`, which has only 1 argument freq and called it in `__eq__ `instead of calling `to_offset` with 2 arguments.

After rebuilding and installing pandas dev environment I checked some examples.

Looks like for Period it works correctly for both ‘M’ and ‘ME’ `pd.Period('2012-1-1', freq='ME')`

But for period_range and PeriodIndex
```
pd.period_range(start='2017-01-01', end='2017-05-01', freq='M')
pd.PeriodIndex(year=[2000, 2002], freq='M')
```
I got an error “TypeError: to_offset() takes exactly 2 positional arguments (1 given)”